src/lib/core/a11y/focus-trap.ts

```diff
@@ -114,4 +158,55 @@ export class FocusTrap {
 
     return null;
   }
+
+  private _createAnchor(): HTMLElement {
+    let anchor = document.createElement('div');
+    anchor.tabIndex = 0;
+    anchor.classList.add('cdk-visually-hidden');
+    anchor.classList.add('cdk-focus-trap-anchor');
+    return anchor;
+  }
+}
+
+
+/**
+ * Service that allows easy instantiation of focus traps.
+ */
+@Injectable()
+export class FocusTrapService {
```

Comment:
I went with the injectable so we don't have to worry about passing the injected `InteractivityChecker` and `NgZone` every time we need to instantiate a focus trap.